lib/ddg-atb.js

```diff
@@ -16,7 +16,8 @@
 
 "use strict";
 
-const {Cu} = require("chrome");
+const {Cu, CC} = require("chrome");
+const XMLHttpRequest = CC("@mozilla.org/xmlextras/xmlhttprequest;1");
```

Comment:
@jdorweiler @zachthompson As I said in the other comment, this is not that important as both of them will most probably get deprecated.

The only reason why I brought this up is that the higher level API has a higher chance of still working even after the deprecation deadline.

Just something to keep in mind, I do not think it is necessary to redo this whole PR, especially if parts of the code can be reused from the Chrome extension.